nondex-common/src/main/java/edu/illinois/nondex/shuffling/ControlNondeterminism.java

```diff
@@ -95,6 +111,33 @@ public static Configuration getConfiguration() {
 
     }
 
+    private static void initializeNondex() {
+        // TODO (XLXL) Temporary solution to prevent JVM crash. It's not safe to invoke isBooted()
+        //  or getProperty before saveAndRemoveProperties in System.initPhase1 is done
+        if (System.out == null) {
+            return;
+        }
+
+        if (Utils.checkJDKBefore8()) {
+            if (nondex == null) {
+                nondex = NonDex.getInstance();
+            }
+            return;
+        }
+
+        try {
+            Method isBooted = Class.forName("jdk.internal.misc.VM").getDeclaredMethod("isBooted");
+            if ((Boolean)isBooted.invoke(null) && nondex == null && !isCreatingNonDex) {
+                isCreatingNonDex = true;
+                // Call getStackTrace here to bypass exception "can't initialize class StackTraceElement$HashedModules"
+                Thread.currentThread().getStackTrace();
+                nondex = NonDex.getInstance();
+            }
+        } catch (Exception ex) {
```

Comment:
Is there some more specific exception we can catch here as to ensure it's an issue with the reflection trying to load a class?